src/overlays/actors/ovl_Demo_Gj/z_demo_gj.c

```diff
@@ -96,241 +84,1326 @@ static ColliderCylinderInitType1 sCylinderInit = {
+        osSyncPrintf("Demo_Gj_Search_Boss_Ganon %d:ガノン発見出来ず\n", this->dyna.actor.params);
+        return false;
+    }
+    //! @bug: Missing return value when `this->ganon` is already set.
+}
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Gj/func_8097BD70.s")
+void DemoGj_InitCommon(DemoGj* this, GlobalContext* globalCtx2, CollisionHeader* header) {
+    GlobalContext* globalCtx = globalCtx2;
+    Actor* actor = &this->dyna.actor;
+    DynaCollisionContext* colCtx_dyna = &globalCtx->colCtx.dyna;
+    CollisionHeader* newHeader;
+
+    if (header != NULL) {
+        Actor_ProcessInitChain(actor, sInitChain);
+        DynaPolyActor_Init(&this->dyna, DPM_UNK);
+        newHeader = NULL;
+        CollisionHeader_GetVirtual(header, &newHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, colCtx_dyna, actor, newHeader);
+    }
+}
+
+s32 DemoGj_InitSetIndexes(DemoGj* this, GlobalContext* globalCtx, s32 updateIndex, s32 drawIndex,
+                          CollisionHeader* header) {
+    if (!DemoGj_IsSceneInvalid()) {
+        this->updateIndex = updateIndex;
+        this->drawIndex = drawIndex;
+        DemoGj_InitCommon(this, globalCtx, header);
+        return 1; // return true; (?)
+    }
+    Actor_Kill(&this->dyna.actor);
+    return 0; // return false; (?)
+}
+
+void DemoGj_DrawCommon(DemoGj* this, GlobalContext* globalCtx, Gfx* displayList) {
+    if (kREG(0) == 0) {
+        GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+
+        OPEN_DISPS(gfxCtx, "../z_demo_gj.c", 1163);
+
+        func_80093D18(gfxCtx);
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx, "../z_demo_gj.c", 1165),
+                  G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+        gSPDisplayList(POLY_OPA_DISP++, displayList);
+        gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
+
+        CLOSE_DISPS(gfxCtx, "../z_demo_gj.c", 1169);
+    }
+}
+
+void func_8097911C(DemoGj* this, GlobalContext* globalCtx2, Gfx* displayList) {
+    GlobalContext* globalCtx = globalCtx2;
+    GraphicsContext* gfxCtx;
+
+    s16 x = this->rotationVec.x;
+    s16 y = this->rotationVec.y;
+    s16 z = this->rotationVec.z;
+
+    s32 pad; // Needed for matching.
+    Mtx* matrix;
+
+    gfxCtx = globalCtx->state.gfxCtx;
+
+    matrix = Graph_Alloc(gfxCtx, sizeof(Mtx));
+
+    OPEN_DISPS(gfxCtx, "../z_demo_gj.c", 1187);
+
+    Matrix_Push();
+    Matrix_RotateRPY(x, y, z, MTXMODE_APPLY);
+    Matrix_ToMtx(matrix, "../z_demo_gj.c", 1193);
+    Matrix_Pull();
+
+    func_80093D18(gfxCtx);
+
+    gSPMatrix(POLY_OPA_DISP++, matrix, G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, displayList);
+    gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
+
+    CLOSE_DISPS(gfxCtx, "../z_demo_gj.c", 1201);
+}
+
+// TODO: find a better name
+void DemoGj_Reflect(DemoGj* this, GlobalContext* globalCtx) {
```

Comment:
Ive mentioned it a few times in this review, but as a general rule of thumb, if you arent confident in a name it is usually a better idea to just leave it unnamed than to try and make something kind of work